### PR TITLE
fix(telemetry): pass signup_timestamp on org_count identify

### PR DESCRIPTION
**Pull request by seanoliver**
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

## What kind of change does this PR introduce?

Bug fix.

## What is the current behavior?

The `dataApiRevokeOnCreateDefault` experiment ([GROWTH-853](https://linear.app/supabase/issue/GROWTH-853)) flipped to 5% rollout on 2026-05-14 22:59 UTC. First-day diagnostic on cohort-filtered exposures showed treatment share at 0.87%, well below the 5% target — about 5× under-bucketed.

Diagnosis: the two audience properties (`org_count` and `signup_timestamp`) were being set on two separate `posthog.identify` calls. Identify #2's `/decide` request body only carried `org_count`. The server-side person record didn't yet have `signup_timestamp` persisted from identify #1 (ingestion is async), so the audience filter failed and the flag returned `false` by default. About 74% of cohort users hit this race.

Pinned down by isolating filters: `org_count = 1` alone vs `org_count = 1 AND signup_timestamp >= rollout` gives the same treatment count (188 vs 190) but the control count halves (28,626 → 14,075). Same numerator, smaller denominator → confirms ~14,500 users had `org_count` matched at flag-evaluation time but `signup_timestamp` not yet landed server-side.

## What is the new behavior?

The studio `Telemetry` component's `org_count` identify now also passes `signup_timestamp` (sourced from `user.created_at`). That makes the `/decide` request body self-sufficient for the audience filter — the server can match `org_count = 1 AND signup_timestamp >= rollout` against request `person_properties` regardless of whether the previous identify's `$set` has been persisted yet.

Kept the existing `signup_timestamp` set in `useTelemetryIdentify` in `packages/common/`. That hook runs in non-studio apps too (www, docs, cms) and they still need `signup_timestamp` set for cross-app analytics.

## Additional context

Validation plan: rerun the cohort-filtered exposure-by-variant query in PostHog 24h post-deploy, filtered to users who signed up after the deploy timestamp. Treatment share should approach 5% vs 0.87% pre-fix. If it doesn't, the assumption about `/decide` honoring request-body `person_properties` is wrong and we need to dig further.

Tracking ticket: [GROWTH-858](https://linear.app/supabase/issue/GROWTH-858).
Live PostHog readout: https://eu.posthog.com/project/34344/insights/fBk4AZ1K.

## Summary by CodeRabbit

* **Chores**
  * Telemetry now records organization count and user signup timestamp together; user identification is re-sent only when those values change to improve analytics accuracy and audience matching.
* **Tests**
  * Added tests validating telemetry identification behavior, deduping across renders, and edge cases when signup timestamp or organization data are missing or become available.

[Review Change Stack](https://app.coderabbit.ai/change-stack/supabase/supabase/pull/46005)

### CODE REVIEW

File: apps/studio/lib/telemetry.tsx [hunk 1]
```diff
@@ -41,7 +41,10 @@ export function Telemetry() {
     const last = lastSentRef.current
     if (last?.userId === user.id && last.orgCount === orgCount) return
     lastSentRef.current = { userId: user.id, orgCount }
-    posthogClient.identify(user.id, { org_count: orgCount })
+    posthogClient.identify(user.id, {
+      org_count: orgCount,
+      ...(user.created_at && { signup_timestamp: user.created_at }),
+    })
   }, [user?.id, organizations])
```
[the remark this answers is not in the document]
Comment (seanoliver) [reply apps/studio/lib/telemetry.tsx]: Good catch, applied the suggested fix in 4f9ee14. Added `user?.created_at` to the effect deps and `signupTimestamp` to the dedup ref so a late-arriving `created_at` triggers a re-run with the full audience properties.

In practice `created_at` is on the session JWT and lands with `user.id`, but the code shouldn't rely on that ordering — defensive against any future auth refactor that splits the parse.

**Review by awaseem — APPROVED** [review]
